src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,46 +509,190 @@ public int getID()
+                case "diamonds" : this.triggerAchievement(ModAchievements.achievementMine500Diamonds);
+                    break;
+                case "huts"     : this.triggerAchievement(ModAchievements.achievementBuild500Huts);
+                    break;
+                case "fish"     : this.triggerAchievement(ModAchievements.achievementCatch500Fish);
+                    break;
+                case "wheat"    : this.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+                    break;
+                case "potatoes" : this.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+                    break;
+                case "carrots"  : this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+                    break;
+                case "saplings" : this.triggerAchievement(ModAchievements.achievementPlant500Saplings);
+                    break;
+                case "trees"    : this.triggerAchievement(ModAchievements.achievementFell500Trees);
+                    break;
+            }
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIFTH)
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIFTH)
         {
-            this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+            switch (statistic)
+            {
+                case "mobs"     : this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+                    break;
+                case "ores"     : this.triggerAchievement(ModAchievements.achievementMine1000Ores);
+                    break;
+                case "diamonds" : this.triggerAchievement(ModAchievements.achievementMine1000Diamonds);
+                    break;
+                case "huts"     : this.triggerAchievement(ModAchievements.achievementBuild1000Huts);
+                    break;
+                case "fish"     : this.triggerAchievement(ModAchievements.achievementCatch1000Fish);
+                    break;
+                case "wheat"    : this.triggerAchievement(ModAchievements.achievementHarvest1000Wheat);
+                    break;
+                case "potatoes" : this.triggerAchievement(ModAchievements.achievementHarvest1000Potatoes);
+                    break;
+                case "carrots"  : this.triggerAchievement(ModAchievements.achievementHarvest1000Carrots);
+                    break;
+                case "saplings" : this.triggerAchievement(ModAchievements.achievementPlant1000Saplings);
+                    break;
+                case "trees"    : this.triggerAchievement(ModAchievements.achievementFell1000Trees);
+                    break;
+            }
         }
     }
 
+
     /**
-     * get the amount of killed mobs.
-     *
-     * @return amount of mobs killed
+     * Get the amount of statistic.
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public int getKilledMobs()
+    public int getStatisticAmount(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "getStatisticAmount" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)